### PR TITLE
Added Embedder and EmbedderRuCLIP classes

### PR DESCRIPTION
**Pull request by keller-kirill**
Added Embedder and EmbedderRuCLIP classes. Currently EmbedderRuCLIP is based on pytorch RuCLIP. 

Added unit tests for the class.

### CODE REVIEW

File: embedder.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    def encode_text(self, text):
+        pass
+
+    @abc.abstractmethod
+    def encode_imgs(self, imgs):
+        pass
+
+    def cos(self, text_emb: np.ndarray, img_emb: np.ndarray) -> Number:
+        """
+        Returns cos similarity between text_emb and img_emb
```
Comment (submaps) [inline embedder.py:28]: Unnecessary mention about `text_emb` and `img_emb`, it compares just 2 embeddings

File: embedder.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        :param text_emb: 1D tensor
+        :param img_emb: 1D tensor
+        :return: cos similarity (Number)
+        """
+        return np.dot(text_emb, img_emb)
```
Comment (submaps) [inline embedder.py:33]: It's like a dot product, not cos distance

File: tests/test_embedder.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+def test_cos():
+    embedder = EmbedderRuCLIP()
+
+    text_emb = np.array([0, 1])
+    img_emb = np.array([0, 1])
+    assert embedder.cos(text_emb, img_emb) == 1
+
+    text_emb = np.array([0, 1, 0, 1, 3, 5, 6, 1])
+    img_emb = np.array([7, 3, 5, 4, 2, 5, 9, 2])
+    assert embedder.cos(text_emb, img_emb) == 94
```
Comment (submaps) [inline tests/test_embedder.py:35]: Related comment to cos func, why cos is 94?

File: tests/test_embedder.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import pytest
```
Comment (submaps) [inline tests/test_embedder.py:1]: I suggest to change `tests` to `test` directory name.

File: tests/test_embedder.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import numpy as np
+from PIL import Image
+
+from embedder import EmbedderRuCLIP
+
+def test_encode_text():
```
Comment (submaps) [inline tests/test_embedder.py:7]: use unittest structure, using `class TestEmbedderRuCLIP(unittest.TestCase)` and then `test_encode_imgs(self)`, and `self.asserEqual` methods

File: embedder.py (hunk 1 above)
Comment (keller-kirill) [reply embedder.py:28]: Ok

File: embedder.py (hunk 2 above)
Comment (keller-kirill) [reply embedder.py:33]: That's right. RuCLIP outputs embeddings with norm 1, so I was thinking to simplify the code. But I will add division by norms product, it costs nothing.

File: tests/test_embedder.py (hunk 3 above)
Comment (keller-kirill) [reply tests/test_embedder.py:35]: Indeed... It was supposed to accept vectors with norm 1. Will be fixed with previous fix.

File: tests/test_embedder.py (hunk 5 above)
Comment (keller-kirill) [reply tests/test_embedder.py:7]: ok

**Review by submaps — APPROVED** [review]

File: embedder.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    def encode_text(self, text: str) -> np.ndarray:
+        """
+        Returns text latent of the text input
+        :param text:
+        :return:
+        """
+        classes = [text, ]
+        with torch.no_grad():
+            text_latent = self.predictor.get_text_latents(classes)
+        return text_latent.cpu().detach().numpy()
```
[the remark this answers is not in the document]
Comment (keller-kirill) [reply embedder.py:57]: Do you mean making something like tonumpy(text_latent) ?

**Pull request merged**